Web scraping (#204)
NoroffFEU/development-platforms: src/content/docs/data-acquisition/web-scraping-with-AI.md

- **CAPTCHA Solving** Some AI Tools can recognise and tackle CAPTCHA challenges
- **User Behaviour Mimicking** AI can simulate human browsing behaviors aids in avoiding detection.
- **No-code / Low-code Platforms** Many AI Scraping tools require minimal coding, making web scraping accessible to a broader audience.

### Drawbacks of AI Tools

- **Length Limit** A common problem when using AI to scrape and process data is the query length limitation. the ChatGPT-3.5 for example has a limit of 4096 tokens per input.
- **AI can be wrong** AI's inability to think critically can lead to incorrect interpretations

### Example of AI tools for web scraping:

- **ChatGpt:** ChatGPT is free, and can create the script and analyse the data with the url provided by the user.
- **AIWebscraper Pro:** Utilizes AI algorithms for fast, accurate data extraction, real-time adaptation to changing website structures, and integration with various databases.
- **Browse AI:** Extract data from any website and turn it into a spreadsheet or an API with No-Code

## Example of web scraping using Chat GPT

> For his case study i decided to web scrape [Noroff staff member site](https://www.noroff.no/en/contact/staff/53-academic)

Reviewer: Fix the tone and make it objective
```suggestion
For this case study, let us web scrape [Noroff staff member site](https://www.noroff.no/en/contact/staff/53-academic)
```

Author: this is not fixed

> First i found and copied the css selector containing the teachers names and titles.

Reviewer: Avoid the use of the word "I" as it makes the sentence sound subjective. Make it objective

Author: I have now rewritten the text in the example.
sry it took so long, i been offline do to being sick
